src/test/java/lambda/part2/exercise/Exercise2.java

```diff
@@ -0,0 +1,98 @@
+class Exercise2 {
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstName() {
+        // TODO предикат Person -> boolean, проверяющий что имя и фамилия человека не пусты
+        Predicate<Person> validate = person -> person.getFirstName().length() != 0 && person.getLastName().length() != 0;
+
+        assertThat(validate.test(new Person("Алексей", "Доренко", 40)), is(true));
+        assertThat(validate.test(new Person("Николай", "", 30)), is(false));
+        assertThat(validate.test(new Person("", "Мельников", 20)), is(false));
+    }
+
+    // TODO метод (Person -> boolean) -> (Person -> boolean)
+    // TODO - возвращает новый предикат, являющийся отрицанием исходного
+    // TODO - при реализации использовать логический оператор !
+    private Predicate<Person> negateUsingLogicalOperator(Predicate<Person> predicate) {
+        return person -> !predicate.test(person);
+    }
+
+    // TODO метод (Person -> boolean, Person -> boolean) -> (Person -> boolean)
+    // TODO - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
+    // TODO - при реализации использовать логический оператор &&
+    private Predicate<Person> andUsingLogicalOperator(Predicate<Person> left, Predicate<Person> right) {
+        return person -> left.test(person) && right.test(person);
+    }
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
+        Predicate<Person> personHasEmptyFirstName = person -> person.getFirstName().length() == 0;
+        Predicate<Person> personHasEmptyLastName = person -> person.getLastName().length() == 0;
+
+        Predicate<Person> personHasNotEmptyFirstName = negateUsingLogicalOperator(personHasEmptyFirstName);
+        Predicate<Person> personHasNotEmptyLastName = negateUsingLogicalOperator(personHasEmptyLastName);
+
+        Predicate<Person> personHasNotEmptyLastNameAndFirstName = andUsingLogicalOperator(personHasNotEmptyFirstName, personHasNotEmptyLastName);
+
+        assertThat(personHasNotEmptyLastNameAndFirstName.test(new Person("Алексей", "Доренко", 40)), is(true));
+        assertThat(personHasNotEmptyLastNameAndFirstName.test(new Person("Николай", "", 30)), is(false));
+        assertThat(personHasNotEmptyLastNameAndFirstName.test(new Person("", "Мельников", 20)), is(false));
+    }
+
+    // TODO метод (T -> boolean) -> (T -> boolean)
+    // TODO - возвращает новый предикат, являющийся отрицанием исходного
+    // TODO - при реализации использовать логический оператор !
+    private <T> Predicate<T> negate(Predicate<T> predicate) {
+        return person -> !predicate.test(person);
+    }
+
+    // TODO метод (T -> boolean, T -> boolean) -> (T -> boolean)
+    // TODO - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
+    // TODO - при реализации использовать логический оператор &&
+    private <T> Predicate<T> and(Predicate<T> left, Predicate<T> right) {
+        return person -> left.test(person) && right.test(person);
```

Comment:
It will be better to use here and in the `negate(Predicate<T> predicate)` method above `t` or something like that instead of `person`, because actually it can be any type, not only Person :)